blocks/ad-taboola-block/features/ad-taboola/default.jsx

```diff
@@ -0,0 +1,162 @@
+    document.getElementsByTagName('script')[0],
+    '//cdn.taboola.com/libtrc/${publisherID}/loader.js',
+    'tb_loader_script');
+    if(window.performance && typeof window.performance.mark == 'function')
+      {window.performance.mark('tbl_ic');}
+  `
+);
+
+const taboolaFlusher = () => (
+  `
+  window._taboola = window._taboola || [];
+  _taboola.push({flush: true});
+  `
+);
+
+@Consumer
+class AdTaboola extends Component {
+  constructor(props) {
+    super(props);
+
+    const { arcSite } = this.props;
+    this.publisherID = getProperties(arcSite).taboolaPublisherId;
+    const { customFields: { placement, mode, container } = {} } = props;
+
+    this.state = {
+      placement,
+      mode,
+      container,
+    };
+  }
+
+  componentDidMount() {
+    if (!this.canRender()) {
+      return;
+    }
+
+    this.insertLoader();
+    this.insertFlusher();
+  }
+
+  appendScript = (name, container, sourceResolver) => {
+    const script = document.createElement('script');
+    script.id = name;
+    script.async = true;
+    script.type = 'text/javascript';
+    script.innerHTML = sourceResolver();
+
+    container.appendChild(script);
+  }
+
+  canRender() {
+    const { placement, mode, container } = this.state;
+    return (this.publisherID && placement && mode && container);
+  }
+
+  insertLoader() {
+    const loader = document.getElementById('tbl-loader');
+    if (loader) {
+      return;
+    }
+    const head = document.getElementsByTagName('head')[0];
+    if (!head) {
+      return;
+    }
+
+    this.appendScript('tbl-loader', head, () => taboolaLoader(this.publisherID));
+  }
+
+  insertFlusher() {
+    const flusher = document.getElementById('tbl-flusher');
+    if (flusher) {
+      return;
+    }
+    const body = document.getElementsByTagName('body')[0];
+    if (!body) {
+      return;
+    }
+
+    this.appendScript('tbl-flusher', body, taboolaFlusher);
+  }
+
+  render() {
+    if (!this.canRender()) {
+      return null;
+    }
+    const { isAdmin } = this.props;
+    const { placement, mode, container } = this.state;
+
+    if (isAdmin) {
+      return (
+        <>
+          <div
+            className="tbl-wrapper"
+            style={{
+              backgroundColor: 'papayawhip',
+              padding: '20px',
+            }}
+          >
+            <small>Taboola widget&nbsp;</small>
+            <strong>{placement}</strong>
+          </div>
+          <hr />
+        </>
+      );
+    }
+
+    const scriptString = `
+      window._taboola = window._taboola || [];
+      _taboola.push({
+        mode: '${mode}',
+        container: '${container}',
+        placement: '${placement}',
+        target_type: 'mix'
+      });
+    `;
+
+    return (
+      <>
+        <div id={`${container}`} />
+        <hr />
+        <script type="text/javascript" dangerouslySetInnerHTML={{ __html: scriptString }} />
+      </>
+    );
+  }
+}
+
+AdTaboola.label = 'Taboola – Arc Block';
```

Comment:
Is Taboola going to be the only ad service we offer?  I just feel 'Taboola – Arc Block' is not descriptive enough...maybe something like 'Taboola Ad Svc – Arc Block' ?

Reply:
I'll add the comment to the ticket to check with Sara, I think make sense.